File: scripts/iprange.pl
#!/usr/bin/perl

# iprange.pl 	A wrapper around iprange.pm module
# INPUT:	ipv4 addr and netmask in CIDR notation
# OUTPUT: 	lowest/highest addrs in input addr's network

BEGIN {push @INC, './perl'}
use strict;
use iprange qw(&iprange);

my %ips = ( cidr => $ARGV[0],
	 low => '',
	 high => '',
);

iprange(\%ips);

if ($ips{low} && $ips{high}) {
    print "$ips{low} $ips{high}\n";
} else {
    print "Error\n";
}

File: iprange/iprange.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <iprange.h>

int main(int argc, char *argv[]) {	
	int r;
	struct cidr_ip_range addrs;

	if (argc != 2) {
	    printf("Bad args\n");
	    exit(-1);
        }

	strlcpy(addrs.ip, argv[1], 100);
	if ((r = ip_range(&addrs)) != 0) {
	    exit(-1);
	}

	printf("%s %s\n", addrs.low, addrs.high);
}
